go/interpreter/lfvm/stack.go

```diff
@@ -11,85 +11,113 @@
 package lfvm
 
 import (
-	"bytes"
 	"fmt"
+	"strings"
 	"sync"
 
 	"github.com/holiman/uint256"
 )
 
-var staticStackBoundary = [NUM_OPCODES]InstructionStack{}
-
-const stackLimit = 1024 // Maximum size of VM stack allowed.
+const maxStackSize = 1024 // Maximum size of VM stack allowed.
 
-type Stack struct {
-	data      [1024]uint256.Int
-	stack_ptr int
+// stack is the 1024-element 256-bit word-wide stack used by the VM.
+// It is a fixed-size stack to prevent memory reallocation during execution.
+// Check boundaries are not checked. Users of the stack must prevent over- and
+// underflow situations.
+//
+// Each stack consumes 1024 * 32 bytes = 32KB of memory. Thus, creating and
+// destroying stacks could incur significant overhead. To mitigate this, a
+// stack pool is provided to reuse stack instances. To obtain an empty stack
+// from the pool, use NewStack(). To return a stack to the pool, use
+// ReturnStack(s).
+//
+// Example usage:
+//
+//	s := NewStack()
+//	defer ReturnStack(s)
+//	<use the stack in your local scope>
+//
+// The stack is not thread-safe. NewStack() and ReturnStack() are thread-safe.
+type stack struct {
+	data         [maxStackSize]uint256.Int
+	stackPointer int
```

Comment:
since you are renaming, do you think `stackHeight` or `stackSize` could be better names for this field ?

Reply:
I think the concept of a `stackPointer` is a well established idea and the name is matching the role it plays here.